extensions/amp-viz-vega/0.1/amp-viz-vega.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isJsonScriptTag, childElementsByTag} from '../../../src/dom';
+import {isExperimentOn} from '../../../src/experiments';
+import {tryParseJson} from '../../../src/json';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {dev, user} from '../../../src/log';
+import {vsyncFor} from '../../../src/vsync';
+
+/** @const */
+const EXPERIMENT = 'amp-viz-vega';
+
+export class AmpVizVega extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    user.assert(isExperimentOn(this.win, EXPERIMENT),
+        `Experiment ${EXPERIMENT} disabled`);
+
+    /** @private {?JSONType} */
+    this.data_ = null;
+
+    /** @private {?string} */
+    this.inlineData_ = this.getInlineData_();
+
+    /** @private {?string} */
+    this.src_ = this.element.getAttribute('src');
+
+    user.assert(this.inlineData_ || this.src_,
+        '%s: neither `src` attribute nor a ' +
+        'valid <script type="application/json"> child was found for Vega data.',
+        this.getName_());
+
+    user.assert(!(this.inlineData_ && this.src_),
+        '%s: both `src` attribute and a valid ' +
+        '<script type="application/json"> child were found for Vega data. ' +
+        'Only one way of specifying the data is allowed.',
+        this.getName_());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.initialize_();
+    return this.loadData_().then(() => this.renderGraph_());
+  }
+
+  /**
+   * Create the vega container.
+   * Called lazily in the first `#layoutCallback`.
+   */
+  initialize_() {
+    if (this.container_) {
```

Comment:
Since you haven't defined an `#unlayoutCallback`, this will only ever be called once, making this check useless.